Modix.Services/Quote/MessageLinkBehavior.cs

```diff
@@ -0,0 +1,94 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using Discord;
+using Discord.WebSocket;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+
+namespace Modix.Services.Quote
+{
+    public class MessageLinkBehavior : BehaviorBase
+    {
+        private static readonly Regex Pattern = new Regex(
+            @"https://(canary\.)?discordapp\.com/channels/(?<GuildId>\d+)/(?<ChannelId>\d+)/(?<MessageId>\d+)",
+            RegexOptions.Compiled | RegexOptions.IgnoreCase | RegexOptions.CultureInvariant);
+
+        public MessageLinkBehavior(DiscordSocketClient discordClient, IServiceProvider serviceProvider)
+            : base(serviceProvider)
+        {
+            DiscordClient = discordClient;
+            Log = serviceProvider.GetRequiredService<ILogger<MessageLinkBehavior>>();
+        }
+
+        private DiscordSocketClient DiscordClient { get; }
+
+        private ILogger<MessageLinkBehavior> Log { get; }
+
+        protected internal override Task OnStartingAsync()
+        {
+            DiscordClient.MessageReceived += OnMessageReceived;
+
+            return Task.CompletedTask;
+        }
+
+        protected internal override Task OnStoppedAsync()
+        {
+            DiscordClient.MessageReceived -= OnMessageReceived;
+
+            return Task.CompletedTask;
+        }
+
+        private async Task OnMessageReceived(SocketMessage message)
+        {
+            if (!(message is SocketUserMessage userMessage) ||
+                !(userMessage.Author is SocketGuildUser || userMessage.Author.IsBot) ||
```

Comment:
Having trouble parsing this line. Is this line supposed to be `!(userMessage.Author is SocketGuildUser) || userMessage.Author.IsBot ||`?

Reply:
This evaluates to `false` if the message author is not a `SocketGuildUser` or if the author is a `SocketGuildUser` but is also bot. I can refactor it to try to make it more clear.

Reply:
Is [this](https://github.com/discord-csharp/MODiX/pull/221/commits/3ec9a9d62b69230d5d7ebbd3bf8333f996ac378c) better?

Reply:
Much clearer. So simple that even I can understand it :)